opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -99,6 +101,30 @@ public function index() {
         $this->data['rss_feed_title'] = $this->language->get('rss_feed_title');
         $this->data['rss_feed_text'] = $this->language->get('rss_feed_text');
         $this->data['rss_feed_url'] = $this->config->get('config_url') . 'index.php?route=smailyforopencart/rss';
+
+        // Layout text and buttons
+        $this->data['entry_layout'] = $this->language->get('entry_layout');
+        $this->data['entry_position'] = $this->language->get('entry_position');
+        $this->data['entry_status'] = $this->language->get('entry_status');
+        $this->data['entry_sort_order'] = $this->language->get('entry_sort_order');
+        $this->data['text_content_top'] = $this->language->get('text_content_top');
```

Comment:
Should I keep the naming scheme of entry_foo for posterity?

Reply:
If `entry_...` makes sense in the context, then yes. But to me `entry_text_content_top` and `entry_content_top` do not make sense, because I do not understand where these variables are used or what for.

Reply:
I am still not happy with the variables. For example with customer synchronization and RSS feed variables I can easily understand what these are used for by looking at the prefix (`customer_sync_` and `rss_feed_` respectively). On the other hand with `entry_title_` I do not know if it is used by customer synchronization, RSS feed or by something completely different (even the comment before this code block says "Layout text and buttons". What is a layout?

I am so adamant about naming things because it will make life easier in few months time when you (or somebody else) needs to make changes in the code.

Note! You do not need to have `$this->data[]` variable name be equal to `$this->language->get()` argument name.

Reply:
I don't know what to rename them to, or even what that code is called. People on OC forums call it  `module layout setting` or some variable of it. Maybe make it `module layout linking`? Open to suggestions

Reply:
I would go with:
```
table_header_layout_text
table_header_position_text
table_header_status_text
table_header_sort_order_text

optin_form_position_top_text
optin_form_position_bottom_text
optin_form_position_left_text
optin_form_position_right_text

button_add_module_text
button_remove_module_text
```

If `table_header_` feels to generic (or conflicts with other variables) you can toss a `optin_form_` prefix to it.

Reply:
Looking at this, it's much clearer. Will add in next commit.